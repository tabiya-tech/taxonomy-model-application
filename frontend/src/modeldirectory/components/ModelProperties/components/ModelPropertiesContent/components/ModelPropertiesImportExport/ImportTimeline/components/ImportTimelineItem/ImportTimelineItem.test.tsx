describe("ImportTimelineItem", () => {
  describe("render tests", () => {
    test.todo("should render with provided importProcessState")
      // GIVEN an importProcessState
      // WHEN the component is rendered
      // THEN expect no errors or warning to have occurred
      // AND the component to be displayed
      // AND the importProcessStateContent component to be displayed
      // AND the ImportProcessStateContent to have been called with the correct props
      // AND the timelineSeparator component to be displayed
      // AND the timelineDot component to be displayed
      // AND the TimelineDot to have been called with the correct props
      // AND the timestamp to be displayed
      // AND the TextPropertyField to have been called with the correct props
      // AND the timelineOppositeContent component to be displayed
      // AND the TimelineOppositeContent to have been called with the correct props
      // AND the component to match the snapshot
  });
});

