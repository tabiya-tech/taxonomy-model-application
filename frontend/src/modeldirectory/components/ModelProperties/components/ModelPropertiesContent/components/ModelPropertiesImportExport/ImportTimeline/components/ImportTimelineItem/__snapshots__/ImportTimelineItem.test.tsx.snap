// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImportTimelineItem render tests should render with fallback text when importProcessState.createdAt is not provided 1`] = `
<li
  class="MuiTimelineItem-root MuiTimelineItem-positionRight css-ny8qih-MuiTimelineItem-root"
  data-testid="import-timeline-item-920424ec-c21e-4673-a127-6c92451a38dc"
>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionRight css-1jlt6h8-MuiTypography-root-MuiTimelineOppositeContent-root"
    data-testid="import-timeline-opposite-content-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 css-17ygoe0-MuiTypography-root"
      data-testid="import-timeline-fallback-text-920424ec-c21e-4673-a127-6c92451a38dc"
    >
      Import has not started yet
    </p>
  </div>
  <div
    class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
    data-testid="import-timeline-separator-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-d8mwxk-MuiTimelineDot-root"
      data-testid="import-timeline-dot-920424ec-c21e-4673-a127-6c92451a38dc"
    />
    <span
      class="MuiTimelineConnector-root css-itego1-MuiTimelineConnector-root"
      data-testid="import-timeline-connector-920424ec-c21e-4673-a127-6c92451a38dc"
    />
  </div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-p3xido-MuiTypography-root-MuiTimelineContent-root"
    data-testid="import-timeline-content-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <div
      data-testid="import-process-state-content-333ab3d5-61a7-4169-8af9-01fe57ed5673"
    >
      ImportProcessStateContent
    </div>
  </div>
</li>
`;

exports[`ImportTimelineItem render tests should render with provided importProcessState 1`] = `
<li
  class="MuiTimelineItem-root MuiTimelineItem-positionRight css-ny8qih-MuiTimelineItem-root"
  data-testid="import-timeline-item-920424ec-c21e-4673-a127-6c92451a38dc"
>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineOppositeContent-root MuiTimelineOppositeContent-positionRight css-1jlt6h8-MuiTypography-root-MuiTimelineOppositeContent-root"
    data-testid="import-timeline-opposite-content-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ghakyl-MuiTypography-root"
      data-testid="import-timeline-opposite-content-created-at-920424ec-c21e-4673-a127-6c92451a38dc"
    >
      foo-date-user-friendly-formatted
    </p>
  </div>
  <div
    class="MuiTimelineSeparator-root css-1fe7e6p-MuiTimelineSeparator-root"
    data-testid="import-timeline-separator-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <span
      class="MuiTimelineDot-root MuiTimelineDot-filled MuiTimelineDot-filledPrimary css-d8mwxk-MuiTimelineDot-root"
      data-testid="import-timeline-dot-920424ec-c21e-4673-a127-6c92451a38dc"
    />
    <span
      class="MuiTimelineConnector-root css-itego1-MuiTimelineConnector-root"
      data-testid="import-timeline-connector-920424ec-c21e-4673-a127-6c92451a38dc"
    />
  </div>
  <div
    class="MuiTypography-root MuiTypography-body1 MuiTimelineContent-root MuiTimelineContent-positionRight css-p3xido-MuiTypography-root-MuiTimelineContent-root"
    data-testid="import-timeline-content-920424ec-c21e-4673-a127-6c92451a38dc"
  >
    <div
      data-testid="import-process-state-content-333ab3d5-61a7-4169-8af9-01fe57ed5673"
    >
      ImportProcessStateContent
    </div>
  </div>
</li>
`;
